stix_shifter/stix_translation/src/modules/elastic_ecs/json/to_stix_map.json

```diff
@@ -0,0 +1,228 @@
+      },
+      {
+        "key": "ipv6-addr.value",
+        "object": "src_ip"
+      },
+      {
+        "key": "network-traffic.src_ref",
+        "object": "nt",
+        "references": "src_ip"
+      }
+    ],
+    "port": {
+      "key": "network-traffic.src_port",
+      "object": "nt",
+      "transformer": "ToInteger"
+    },
+    "mac": [
+      {
+        "key": "mac-addr.value",
+        "object": "src_mac"
+      },
+      {
+        "key": "ipv4-addr.resolves_to_refs",
+        "object": "src_ip",
+        "references": "src_mac"
+      },
+      {
+        "key": "ipv6-addr.resolves_to_refs",
+        "object": "src_ip",
+        "references": "src_mac"
+      }
+    ]
+  },
+  "destination": {
+    "ip": [
+      {
+      "key": "ipv4-addr.value",
+      "object": "dst_ip"
+      },
+      {
+        "key": "ipv6-addr.value",
+        "object": "dst_ip"
+      },
+      {
+        "key": "network-traffic.dst_ref",
+        "object": "nt",
+        "references": "dst_ip"
+      }
+    ],
+    "port": {
+      "key": "network-traffic.dst_port",
+      "object": "nt",
+      "transformer": "ToInteger"
+    },
+    "mac": [
+      {
+        "key": "mac-addr.value",
+        "object": "dst_mac"
+      },
+      {
+        "key": "ipv4-addr.resolves_to_refs",
+        "object": "dst_ip",
+        "references": "dst_mac"
+      },
+      {
+        "key": "ipv6-addr.resolves_to_refs",
+        "object": "dst_ip",
+        "references": "dst_mac"
+      }
+    ]
+  },
+  "server": {
+    "ip": [
+      {
+      "key": "ipv4-addr.value",
+      "object": "dst_ip"
+      },
+      {
+        "key": "ipv6-addr.value",
+        "object": "dst_ip"
+      },
+      {
+        "key": "network-traffic.dst_ref",
+        "object": "nt",
+        "references": "dst_ip"
+      }
+    ],
+    "port": {
+      "key": "network-traffic.dst_port",
+      "object": "nt",
+      "transformer": "ToInteger"
+    },
+    "mac": [
+      {
+        "key": "mac-addr.value",
+        "object": "dst_mac"
+      },
+      {
+        "key": "ipv4-addr.resolves_to_refs",
+        "object": "dst_ip",
+        "references": "dst_mac"
+      },
+      {
+        "key": "ipv6-addr.resolves_to_refs",
+        "object": "dst_ip",
+        "references": "dst_mac"
+      }
+    ]
+  },
+  "network": {
+    "transport": {
+      "key": "network-traffic.protocols",
+      "object": "nt",
+      "group" : "True",
+      "transformer": "ToLowercaseArray"
+    },
+    "type": {
+      "key": "network-traffic.protocols",
+      "object": "nt",
+      "group" : "True",
+      "transformer": "ToLowercaseArray"
+    },
+    "protocol": {
+      "key": "network-traffic.protocols",
+      "object": "nt",
+      "group" : "True",
+      "transformer": "ToLowercaseArray"
+    }
+  },
+  "event": {
+    "original": {
+      "key": "artifact.payload_bin",
+      "transformer": "ToBase64"
+    }
+  },
+  "url": {
+    "original": {
+      "key": "url.value"
+    }
```

Comment:
You could add mapping for the ECS `url.domain` field. It would get stored in the `domain.value` STIX attribute. However, since ECS url.domain can also store IP and port numbers, you would need to make sure the value is an actual domain before storing it in the STIX object.

Reply:
url.domain may consist of IP address or domain. I will analyse the code to add new mapping. 
Based on the above suggestion, I will need to add Stix validation for domain name.

Reply:
Updated code to include url.domain. Ref: c6aee8a
Added validation for domain-name.value to check whether value is domain or not. 
 
Please review and let me know comments.